lib/apiservers/engine/backends/container.go

```diff
@@ -689,8 +694,12 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 	if err != nil {
 		switch err := err.(type) {
 		case *containers.ContainerRemoveNotFound:
-			cache.ContainerCache().DeleteContainer(id)
-			return NotFoundError(name)
+			// if this is an autoremove container then it was deleted by the die event
+			// and will be removed from the cache on the destroy event
+			if !vc.HostConfig.AutoRemove {
+				cache.ContainerCache().DeleteContainer(id)
```

Comment:
Is there any way this could happen if events are not up and running yet so we would not remove the container? Can we just make DeleteContainer idempotent so that questions like this are not even applicable?